goupnp.go

```diff
@@ -125,15 +125,29 @@ func DeviceByURL(ctx context.Context, loc *url.URL) (*RootDevice, error) {
 // but should not be changed after requesting clients.
 var CharsetReaderDefault func(charset string, input io.Reader) (io.Reader, error)
 
-// HTTPClient is the client used inside of requestXml for fetching XML from a specific endpoint.
-var HTTPClient = http.DefaultClient
+// contextKey is an unexported type which prevents construction of other contextKeys.
+type contextKey struct{}
```

Comment:
if there's only 1 possible value for this, might as well name it `httpContextKey`.

Then the convention nowadays is to provide the context-wrapping func that does the https://pkg.go.dev/context#WithValue call:

```go
   // WithHTTPClient returns a context wrapping ctx with the HTTP client set to c.
   // c must be non-nil.
   func WithHTTPClient(ctx context.Context, c *http.Client) context.Value {
          return context.WithValue(ctx, httpContextKey{}, c)
   }
```